test/mlir/affine_conversion/core_ops.mlir

```diff
@@ -55,3 +54,82 @@ func @add(%arg0: !ng.tensor<3x2xf32>, %arg1: !ng.tensor<3x2xf32>) -> !ng.tensor<
   %3 = "ng.add"(%2, %2) : (!ng.tensor<3x2xf32>, !ng.tensor<3x2xf32>) -> !ng.tensor<3x2xf32>
   "ng.return"(%3) : (!ng.tensor<3x2xf32>) -> ()
 }
+
+// -----
+
+// Convolution
+// CHECK-LABEL: @convolution
+// Initialization loops
+// CHECK:         affine.for
+// CHECK-NEXT:    affine.for
+// CHECK-NEXT:    affine.for
+// CHECK-NEXT:    affine.for
+// CHECK:         affine.store
+// Convolution loops
+// CHECK:         affine.for %[[a3:.*]] = 0 to 1
+// CHECK:         affine.for %[[a4:.*]] = 0 to 2
+// CHECK:         affine.for %[[a5:.*]] = 0 to 2
+// CHECK:         affine.for %[[a6:.*]] = 0 to 2
+// CHECK:         affine.for %[[a7:.*]] = 0 to 2
+// CHECK:         affine.for %[[a8:.*]] = 0 to 1
+// CHECK:         affine.for %[[a9:.*]] = 0 to 1
+// CHECK:         affine.load %{{.*}}[%[[a4]], %{{.*}}, %[[a8]], %[[a9]]] : memref<2x2x1x1xf32>
+// CHECK:         affine.load %{{.*}}[%[[a3]], %[[a5]], %{{.*}}, {{.*}}] : memref<1x2x2x2xf32>
+// CHECK-NEXT:    mulf
+// CHECK-NEXT:    affine.load %{{.*}}[%[[a3]], %[[a4]], %[[a6]], %[[a7]]] : memref<1x2x2x2xf32>
+// CHECK-NEXT:    %[[v4:.*]] = addf
+// CHECK-NEXT:    affine.store %[[v4]], %{{.*}}[%[[a3]], %[[a4]], %[[a6]], %[[a7]]] : memref<1x2x2x2xf32>
```

Comment:
Showing some kind of structure with the CHECK rules would be great. It would help see the loop nest structure and to which loop the operation belongs to

Reply:
You mean adding indentation ?

Reply:
yes, something that allows us to see the structure

Reply:
This is not resolved

Reply:
What's not resolved about it ? 
`
// CHECK:         affine.for %[[a4:.*]] = 0 to 1
// CHECK:           affine.for %[[a5:.*]] = #[[M2]](%[[v2]]) to #[[M2]](%[[v3]])
// CHECK:             affine.for %[[a6:.*]] = #[[M2]](%[[v0]]) to #[[M2]](%[[v1]])
// CHECK:               affine.for %[[a7:.*]] = 0 to 2
// CHECK:                 affine.for %[[a8:.*]] = 0 to 2
// CHECK:                   affine.for %[[a9:.*]] = 0 to 1
// CHECK:                     affine.for %[[a10:.*]] = 0 to 1
`

Reply:
Are you looking at the latest commit ?